libcalico-go/lib/ipam/ipam.go

```diff
@@ -1805,24 +1806,57 @@ func (c ipamClient) GetAssignmentAttributes(ctx context.Context, addr net.IP) (m
 // has been set, returns a default configuration with StrictAffinity disabled
 // and AutoAllocateBlocks enabled.
 func (c ipamClient) GetIPAMConfig(ctx context.Context) (config *IPAMConfig, err error) {
-	obj, err := c.client.Get(ctx, model.IPAMConfigKey{}, "")
-	if err != nil {
-		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
-			// IPAMConfig has not been explicitly set.  Return
-			// a default IPAM configuration.
-			config = &IPAMConfig{
-				AutoAllocateBlocks: true,
-				StrictAffinity:     false,
-				MaxBlocksPerHost:   0,
+	var obj *model.KVPair
+
+	var retries int
+	maxRetry := 5
+
+	// First, trying to get IPAM config (create if not exists) for 5 times.
```

Comment:
This part of the code is the only change I added to IPAM. It tries to create a default object if not exists.

Reply:
Out of curiosity, how did the IPAM Config creation used to get handled? Did we really just return the default in the absence of anything until someone finally wanted to make changes and saved something different?

Reply:
Currently the only way to create an actual object in the datastore is via calicoctl. It calls `ipam.SetIPAMConfig` function to apply the change.